### PR TITLE
Rake task for changelog updates

### PR DESCRIPTION
**Pull request by amcaplan**
### WHY are these changes introduced?

Saves us the manual task of finding the PR and copypasting the title/number into the changelog.

### WHAT is this pull request doing?

The same thing we do already, just automatically.

Note that no GitHub auth token is required for this task, as it's operating on public data.

If we wanted, we could build this into a GitHub action, but I'm not sure it's worth it.

Note this doesn't work for external contributors yet.

### How to test your changes?

I'll test on this PR (but won't push the change since it's not actually user-facing!)

### Update checklist

N/A
- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [ ] I've left the version number as is (we'll handle incrementing this when releasing).

### CODE REVIEW

**Comment by amcaplan** [comment]
Running the task locally (`bundle exec rake changelog:update`) gives this diff. Looks correct to me.

**Comment by amcaplan** [comment]
Indicating `Fixed` instead of `Added` yields this diff:

Screen Shot 2022-03-27 at 17 54 24

File: test/shopify-cli/changelog_test.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+require "test_helper"
+require "shopify_cli/changelog"
+
+module ShopifyCLI
+  class ChangelogTest < MiniTest::Test
+    def test_changelog_outputs_equal_inputs
+      assert_equal(Changelog.new.full_contents, File.read(Changelog::CHANGELOG_FILE))
```
Comment (amcaplan) [inline test/shopify-cli/changelog_test.rb:8]: This test is really important! It asserts that, without any changes, the parser will output the exact file char-for-char that it took as input. So we can be sure that it's respecting whatever we've done in the past, even when a change is recorded.

File: test/shopify-cli/changelog_test.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+      ### Added
+      * [#5678](https://github.com/Shopify/shopify-cli/pull/5678): Added 5678
+    CHANGELOG
+
+    def test_changelog_outputs_equal_inputs
+      assert_equal(Changelog.new.full_contents, File.read(Changelog::CHANGELOG_FILE))
+    end
+
+    def test_adding_to_existing_section
```
Comment (amcaplan) [inline test/shopify-cli/changelog_test.rb:25]: These tests are intended to add confidence that the changelog is being built correctly, i.e.:

1. Changes are added to the `Unreleased` section
2. Changes are added to the correct section, and a new section is opened if necessary.

**Review by amcaplan — COMMENTED** [review]
I realize that `ShopifyCLI::Changelog` is becoming a junk drawer, but I'm OK with that for now. I don't like the alternative of having many files which aren't really necessary for the CLI. Anyway, I don't see this file growing a lot more, so maybe this is fine.

**Review by gonzaloriestra — APPROVED** [review]
👌

**Pull request merged**